### PR TITLE
feat(analytics): add filtering of bot events in create_pr_metrics and…

### PR DESCRIPTION
**Pull request by Ayyanaruto**
This pull request introduces changes to filter out bot events from pull request metrics and adds corresponding test cases to ensure the correctness of this new functionality. The most important changes include modifying the `create_pr_metrics` function and adding several new test functions.

Filtering bot events:

* [`backend/analytics_server/mhq/service/code/sync/etl_code_analytics.py`](diffhunk://#diff-517b77ebf07ce3d53443b38b360b0bac87cfb171088de3a18857184b1ec15b27L24-R35): Modified the `create_pr_metrics` function to filter out bot events from the pull request events.

New test cases:

* [`backend/analytics_server/tests/service/code/sync/test_etl_code_analytics.py`](diffhunk://#diff-282c361b097e4fa84379df0eb8eae6940aee63d46ca96b3fe9f3593568852e2dR503-R603): Added the `test_create_pr_metrics_filters_bot_events` function to verify that bot events are filtered out correctly.
* [`backend/analytics_server/tests/service/code/sync/test_etl_code_analytics.py`](diffhunk://#diff-282c361b097e4fa84379df0eb8eae6940aee63d46ca96b3fe9f3593568852e2dR503-R603): Added the `test_create_pr_metrics_no_bot_first_response_time` function to ensure that the first response time is calculated correctly when there are no bot events.
* [`backend/analytics_server/tests/service/code/sync/test_etl_code_analytics.py`](diffhunk://#diff-282c361b097e4fa84379df0eb8eae6940aee63d46ca96b3fe9f3593568852e2dR503-R603): Added the `test_create_pr_metrics_no_bot_rework_time` function to verify that the rework time is calculated correctly in the absence of bot events.
* [`backend/analytics_server/tests/service/code/sync/test_etl_code_analytics.py`](diffhunk://#diff-282c361b097e4fa84379df0eb8eae6940aee63d46ca96b3fe9f3593568852e2dR503-R603): Added the `test_create_pr_metrics_no_human_first_response_time` and `test_create_pr_metrics_no_human_rework_time` functions to ensure that metrics are not calculated when only bot events are present.#635 

### Testing
[Youtube](https://youtu.be/uiODehmClzA).

## Summary by CodeRabbit

- **New Features**
	- Improved pull request performance metrics by filtering out events generated by bots, ensuring metrics reflect actual human interactions.
  
- **Tests**
	- Expanded test coverage to verify that only human reviewer events are considered in performance calculations, ensuring accurate timing measurements for review responses and rework.
	- Added comprehensive tests for bot name detection and event filtering, covering common patterns, edge cases, and mixed event scenarios.

### CODE REVIEW

File: backend/analytics_server/mhq/service/code/sync/etl_code_analytics.py [hunk 1]
```diff
@@ -21,7 +21,18 @@ def create_pr_metrics(
...
+        pr_events = [
+            event
+            for event in pr_events
+            if (
+                not event.actor_username.endswith("[bot]")
+                and not (
+                    event.data.get("user")
+                    and event.data.get("user", {}).get("type") == "Bot"
+                )
+            )
```
Comment (samad-yar-khan) [inline backend/analytics_server/mhq/service/code/sync/etl_code_analytics.py:34]: can be contained within a separate function.

File: backend/analytics_server/mhq/service/code/sync/etl_code_analytics.py [hunk 2]
```diff
@@ -21,7 +21,18 @@ def create_pr_metrics(
     ) -> PullRequest:
         if pr.state == PullRequestState.OPEN:
             return pr
-
+        # Filter bots from PR events
+        pr_events = [
```
Comment (samad-yar-khan) [inline backend/analytics_server/mhq/service/code/sync/etl_code_analytics.py:25]: non_bot_pr_events

File: backend/analytics_server/tests/service/code/sync/test_etl_code_analytics.py [hunk 3]
```diff
@@ -500,3 +500,104 @@ def test_rework_cycles_returs_1_for_multiple_approvals():
...
+    human_event = get_pull_request_event(
+        pull_request_id=pr.id,
+        reviewer="human_user",
+        state=PullRequestEventState.APPROVED.value,
+        created_at=t3,
+    )
+    pr_metrics = pr_service.create_pr_metrics(pr, [bot_event, human_event], [])
+    assert "human_user" in pr_metrics.reviewers
+    assert "test-bot[bot]" not in pr_metrics.reviewers
+
```
Comment (samad-yar-khan) [inline backend/analytics_server/tests/service/code/sync/test_etl_code_analytics.py:526]: add test for bot_type user

**Review by samad-yar-khan — CHANGES_REQUESTED** [review]

File: backend/analytics_server/mhq/service/code/sync/etl_code_analytics.py [hunk 4]
```diff
@@ -173,3 +179,9 @@ def get_rework_cycles(
                 rework_cycles += 1
 
         return rework_cycles
+
+    def filter_non_bot_events(
+        self, pr_events: List[PullRequestEvent]
+    ) -> List[PullRequestEvent]:
+        """Filter out events created by bot users using regex patterns."""
+        return [event for event in pr_events if (not is_bot_name(event.actor_username))]
```
Comment (samad-yar-khan) [inline backend/analytics_server/mhq/service/code/sync/etl_code_analytics.py:187]: Just check for null usernames, rest looks good!
        return [
            event
            for event in pr_events
            if event.actor_username is not None
            and (not is_bot_name(event.actor_username))
        ]

**Review by samad-yar-khan — CHANGES_REQUESTED** [review]

File: backend/analytics_server/mhq/service/code/sync/etl_code_analytics.py (hunk 4 above)
Comment (Ayyanaruto) [reply backend/analytics_server/mhq/service/code/sync/etl_code_analytics.py:187]: Will do it now

Comment (Ayyanaruto) [reply backend/analytics_server/mhq/service/code/sync/etl_code_analytics.py]: Done

**Review by samad-yar-khan — APPROVED** [review]
LGTM nice work @Ayyanaruto !

**Comment by Ayyanaruto** [comment]
Thanks, at-last PR merge hogaya 😭
